### PR TITLE
add tensorflow ver. 2.3, added test set split to mnist

### PR DESCRIPTION
**Pull request by kshannon**
The pytorch notebooks failed on pip install, was missing torch.

### CODE REVIEW

**Comment by openfl-helper** [comment]
Can one of the admins verify this patch?

File: openfl-tutorials/Federated_Keras_MNIST_Tutorial.ipynb [hunk 1]
```diff
@@ -14,7 +14,7 @@
    "outputs": [],
    "source": [
     "#Install dependencies if not already installed\n",
-    "!pip install tensorflow mnist"
+    "!pip install tensorflow==2.3 mnist"
```
Comment (alexey-gruzdev) [inline openfl-tutorials/Federated_Keras_MNIST_Tutorial.ipynb:17]: @kshannon  I would recommend to use 2.3.1 version to be consistent with [workspace requirements](https://github.com/intel/openfl/blob/develop/openfl-workspace/keras_cnn_mnist/requirements.txt#L1)
And since you are adding check for intel-tensorflow, let's add some comment like that:
```
"!pip install tensorflow==2.3.1 mnist"
# Or you could use intel-tensorflow
# !pip install intel-tensorflow==2.3.0 mnist"
```

File: openfl-tutorials/Federated_Keras_MNIST_Tutorial.ipynb [hunk 2]
```diff
@@ -69,8 +94,12 @@
     "#Import training and validation images/labels\n",
     "train_images = mnist.train_images()\n",
     "train_labels = to_categorical(mnist.train_labels())\n",
-    "valid_images = mnist.test_images()\n",
-    "valid_labels = to_categorical(mnist.test_labels())\n",
+    "\n",
+    "valid_images = mnist.test_images()[0:8000,:,:]\n",
```
Comment (alexey-gruzdev) [inline openfl-tutorials/Federated_Keras_MNIST_Tutorial.ipynb:98]: I would also suggest adding a variable to handle 8000.

**Comment by alexey-gruzdev** [comment]
Jenkins please retry a build

File: openfl-tutorials/Federated_Keras_MNIST_Tutorial.ipynb (hunk 2 above)
Comment (kshannon) [reply openfl-tutorials/Federated_Keras_MNIST_Tutorial.ipynb]: @alexey-gruzdev  good catch! thanks for the suggestion.

File: openfl-tutorials/Federated_Keras_MNIST_Tutorial.ipynb [hunk 3]
```diff
@@ -10,11 +10,16 @@
...
+   },
    "outputs": [],
    "source": [
-    "#Install dependencies if not already installed\n",
-    "!pip install tensorflow==2.3 mnist"
+    "#Install Tensorflow and MNIST dataset if not installed\n",
+    "!pip install tensorflow==2.3.1 mnist\n",
+    "\n",
+    "#Alternatively you could use the intel-tensorflow build\n",
+    "# !pip install intel-tensorflow==2.3.1 mnist"
```
Comment (grib0ed0v) [inline openfl-tutorials/Federated_Keras_MNIST_Tutorial.ipynb:22]: If I would run it I get the following output:
```
ERROR: Could not find a version that satisfies the requirement intel-tensorflow==2.3.1
ERROR: No matching distribution found for intel-tensorflow==2.3.1
```

File: openfl-tutorials/Federated_Keras_MNIST_Tutorial.ipynb [hunk 4]
```diff
@@ -10,11 +10,16 @@
...
+   },
    "outputs": [],
    "source": [
-    "#Install dependencies if not already installed\n",
-    "!pip install tensorflow mnist"
+    "#Install Tensorflow and MNIST dataset if not installed\n",
+    "!pip install tensorflow==2.3.1 mnist\n",
+    "\n",
+    "#Alternatively you could use the intel-tensorflow build\n",
+    "# !pip install intel-tensorflow==2.3.1 mnist"
```
Comment (alexey-gruzdev) [inline openfl-tutorials/Federated_Keras_MNIST_Tutorial.ipynb:22]: If I would run it I get the following:
```
ERROR: Could not find a version that satisfies the requirement intel-tensorflow==2.3.1
ERROR: No matching distribution found for intel-tensorflow==2.3.1
```

Comment (kshannon) [reply openfl-tutorials/Federated_Keras_MNIST_Tutorial.ipynb:22]: Ack! should be intel-tensorflow==2.3.**0**, I will also try out 2.4.0 real quick to see if there are any issues with the latest build and update accordingly, thanks!

Comment (kshannon) [reply openfl-tutorials/Federated_Keras_MNIST_Tutorial.ipynb:22]: 2.4.0 works, but does throw several dependency errors in the console about numpy and tensorflow-estimator version conflicts. It's probably best for the end user experience to keep intel-tensorflow at 2.3.0 since this tutorial doesn't depend on anything from the latest build.

Comment (psfoley) [reply openfl-tutorials/Federated_Keras_MNIST_Tutorial.ipynb]: Thanks for the contribution @kshannon. Yes, please keep intel-tensorflow 2.3.0 for now. We currently rely on a gRPC version that has conflicts with 2.4.0 (this should be resolved soon).
